min.py

```diff
@@ -135,6 +136,37 @@ def get_min_scipy_version(min_python, min_numpy):
     return df.SciPy_version
 
 
+def find_pkg_mismatches(pkgs, fnames):
+    """
+    Determine if any package version mismatches
+    """
+    pkg_mismatches = []
+
+    for pkg_name, pkg_version in pkgs.items():
```

Comment:
As mentioned in the comment above, if we change the function signature then this `for` loop can be eliminated. In general, we try to avoid having deeply nested loops as it as adds too much burden to the reader to understand the code. One for-loop is good, two nested for-loops is fine, but three or more nested for-loops usually means that we should reconsider the design/abstraction of our function(s)

Reply:
If I remove this for loop how should I return a set of results for a full scan (i.e. across all packages/versions), would this be acceptable if I fix the signature of `find_pkg_mismatches`?

`pkg_mismatches = [find_pkg_mismatches(pkg, version, fnames) for pkg, version in pkgs.items()]
 print(pkg_mismatches)`